tests/integration/test_system_app_creator.py

```diff
@@ -0,0 +1,36 @@
+from .common import random_str
+import time
+
+
+def test_system_app_creator(admin_mc, admin_system_pc, remove_resource):
+    client = admin_mc.client
+    provider_name = random_str()
+    access = random_str()
+    secret = random_str()
+    globaldns_provider = \
+        client.create_global_dns_provider(
+                                         name=provider_name,
+                                         rootDomain="example.com",
+                                         route53ProviderConfig={
+                                             'accessKey': access,
+                                             'secretKey': secret})
+    app = wait_for_system_app(
```

Comment:
I think we should use `remove_resource` immediately after the call to resource creation, the wait_for_system_app could fail before remove_resource is applied.